tools/mongodb/check-mongo.py

```diff
@@ -0,0 +1,55 @@
+import bson
+import sets
+import os
+
+root = "."
+
+constraints = [
+  ("Datasets in volumeTracings should exist", "dataSets.name", "volumes.dataSetName"),
+]
+
+def bson_files(root):
+  for root, dirs, files in os.walk('.', followlinks=True):
+    for file in files:
+      if file.endswith(".bson"):
+        print "Reading data from '%s' ..." % file,
+        yield os.path.join(root, file)
+        print " Done."
+
+def process_document(id, key, doc):
+  if type(doc) == dict:
+    for k in doc:
+      process_document(id, key + "." + k, doc[k])
+  elif type(doc) == list:
+    for v in doc:
+      process_document(id, key + "[]", v)
+  else:
+    if key in keys:
+      if not doc in data[key]:
+        data[key][doc] = []
+      data[key][doc].append(id)
+
+# Initialization
+keys = sets.Set(map(lambda c: c[1], constraints) + map(lambda c: c[2], constraints))
```

Comment:
I don't think you need to import `sets`. `set` should be globally available already